### ISSUE TITLE
zpool add command should refuse to add non-redundant disk configurations to pools without an override option

### ISSUE THREAD

**Issue by karloor** [zfsonlinux/zfs#2486]
At the moment it's embarrassingly easy to do this: 

```
zfs add tank sdn
```

when you mean to do this: 

```
zfs add tank log sdn 
```

@ryao in a IRC chat suggests that adding a non-redundant vdev configuration to a pool like this should only be possible by supplying a flag to force it.

**Comment by adilger** [comment]
It wouldn't be a terrible idea to structure this so zpool reports an error if adding a single-disk VDEV to a pool *that does not yet have any other single-disk VDEVs*, unless something like "--force-single-disk" was used (or answering "yes" at a prompt in an interactive shell).  This is a problem that has hit many people (one more this week), and since there is no easy way to repair this afterward, it shouldn't be so easily done.

By only limiting the error to cases where no other single-disk VDEVs already exist in the pool, this avoids problems for cases where someone is creating a new single-disk test pool, or has previously forced a single disk into the pool and knows that is what they want.

**Comment by behlendorf** [comment]
The original issue here was addressed in master.  The fix will be in 0.7.0.  When adding devices it now verifies that those devices are of a similar replication level.  If they're not you must provide the `- f` option, for example:

```
# zpool create tank mirror vdb vdc
# zpool add tank vdd
invalid vdev specification
use '-f' to override the following errors:
mismatched replication level: pool uses mirror and new vdev is disk
```

**Comment by adilger** [comment]
Overloading "-f" to mean different things (add VDEVs of different Redundancy, add VDEVs on partitions, etc) is not really a good idea. Better to have something like "--force-nonredundant-vdev" or similar.
